src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -1179,3 +1193,85 @@ def build_udtf(
         t = ast.kwargs.add()
         t._1 = k
         build_expr_from_python_val(t._2, v)
+
+
+def build_sproc(
+    ast: proto.StoredProcedure,
+    func: Union[Callable, Tuple[str, str]],
+    return_type: Optional[DataType],
+    input_types: Optional[List[DataType]],
+    sp_name: str,
+    stage_location: Optional[str] = None,
+    imports: Optional[List[Union[str, Tuple[str, str]]]] = None,
+    packages: Optional[List[Union[str, ModuleType]]] = None,
+    replace: bool = False,
+    if_not_exists: bool = False,
+    parallel: int = 4,
+    strict: bool = False,
+    external_access_integrations: Optional[List[str]] = None,
+    secrets: Optional[Dict[str, str]] = None,
+    comment: Optional[str] = None,
+    statement_params: Optional[Dict[str, str]] = None,
+    execute_as: typing.Literal["caller", "owner"] = "owner",
+    source_code_display: bool = True,
+    is_permanent: bool = False,
+    session=None,
+    **kwargs,
+):
+    """Helper function to encode stored procedure parameters (used in both regular and mock StoredProcedureRegistration)."""
+    # TODO: to unparse/reference callables client-side - track them in ast_batch.
```

Comment:
can we resolve this TODO? it should be fairly easy :)
Afaik this here should take core of it
```
 build_proto_from_callable(
        ast.func, func, session._ast_batch if session is not None else None
    )
```

Reply:
Ooh sorry this was an artifact from copying over the code in `build_udf`. Should I get rid of the TODO over there as well?

Reply:
Discussed on slack - removed both TODOs here. Verified that the `id` field of `SpCallable` is set.